dev/PushNotifications/PushBackgroundTaskInstance.h

```diff
@@ -1,26 +1,28 @@
-// Copyright (c) Microsoft Corporation and Contributors.
+﻿// Copyright (c) Microsoft Corporation and Contributors.
 // Licensed under the MIT License.
 
 #include <winrt/Windows.ApplicationModel.background.h>
+#include "PushRawNotification.h"
+#include "PushNotificationDummyDeferral.h"
 
 // Mocks IBackgroundTaskInstance to send raw payloads to packaged apps in
 // PushNotificationBackgroundTask::Run by com activation from PushNotificationsLongRunningProcess
 struct PushBackgroundTaskInstance : winrt::implements<PushBackgroundTaskInstance, winrt::Windows::ApplicationModel::Background::IBackgroundTaskInstance>
 {
     PushBackgroundTaskInstance() {};
-    PushBackgroundTaskInstance(std::wstring const& payload): m_payload(payload) {};
+    PushBackgroundTaskInstance(std::wstring const& payload) : m_rawNotification(winrt::make_self<PushRawNotification>(payload)) {};
 
     winrt::guid InstanceId() { return winrt::guid(); };
     UINT32 SuspendedCount() { return 0; };
     UINT32 Progress() { return 0; };
     UINT32 Progress(UINT32 /* progress */) { return 0; };
     winrt::Windows::ApplicationModel::Background::BackgroundTaskRegistration Task() { return nullptr; };
```

Comment:
This is returning a nullptr too and could result in a crash just like GetDeferral. Should we track through a seperate bug?

Reply:
We don't expose this attribute in PushNotificationReceivedEventArgs so there is no risk in leaving it as nullptr. Similar to SuspendedCount/InstanceId we don't actually use those properties other than to align with the IBackgroundTaskInstance interface.